specification/fist/resource-manager/Microsoft.IoTFirmwareDefense/preview/2023-02-08-preview/workspaces.json

```diff
@@ -0,0 +1,477 @@
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "type": "string",
+          "enum": [
+            "Accepted",
+            "Succeeded",
+            "Canceled",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          },
+          "readOnly": true
+        }
+      },
+      "description": "Workspace properties."
+    },
+    "Workspace": {
+      "type": "object",
+      "description": "Firmware analysis workspace.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WorkspaceProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "WorkspaceUpdateDefinition": {
+      "type": "object",
+      "description": "Firmware analysis workspace.",
+      "properties": {
+        "properties": {
+          "description": "The editable workspace properties.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WorkspaceProperties"
+        }
+      }
+    },
+    "WorkspaceList": {
+      "type": "object",
+      "description": "Return a list of firmware analysis workspaces.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Workspace"
+          },
+          "description": "The list of firmware analysis workspaces."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The uri to fetch the next page of asset."
+        }
+      }
+    },
+    "GenerateUploadUrlProperties": {
+      "type": "object",
+      "description": "Properties for generating an upload URL",
+      "properties": {
+        "firmwareId": {
+          "type": "string",
+          "description": "A unique ID for the firmware to be uploaded."
+        }
+      }
+    }
+  },
+  "parameters": {
+    "WorkspaceCreateBodyParameter": {
+      "name": "workspace",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/Workspace"
+      },
+      "description": "Parameters when creating a firmware analysis workspace.",
+      "x-ms-parameter-location": "method"
+    },
+    "WorkspaceUpdateBodyParameter": {
+      "name": "workspace",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/WorkspaceUpdateDefinition"
+      },
+      "description": "Parameters when updating a firmware analysis workspace.",
+      "x-ms-parameter-location": "method"
+    },
+    "GenerateUploadUrlBodyParameter": {
+      "name": "generateUploadUrl",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/GenerateUploadUrlProperties"
```

Comment:
>GenerateUploadUrlProperties [](http://example.com/codeflow?start=31&length=27)

naming nit, I might call it 'Request' instead of Properties since its just the whole request body/object, and not just the properties, or a set of nested properties